Monitoring in a Box/Ansible/host_vars/mib1.yaml

```diff
@@ -0,0 +1,30 @@
+### VM variables
+## Which image format to use
+# qcow2
+# vmdk
+# raw
+ImageFormat: vmdk
+
+## New VM ID number
+NewVmID: "240"
+
+## New VM name
+NewVmName: "MiB-Host"
+
+## New account username
+NewVmUsername: mib
+
+## VM's IP address
+NewVmIP: 10.0.1.240
+
+# Pve node sudo password
+ansible_become_pass: [key in pve node sudo password]
+
+## New account password must be the same as pve node sudo password
+NewVmPassword: "{{ansible_become_pass}}"
```

Comment:
Actually not true. The Ubuntu template's default setting enables passwordless sudo. This means the Ansible script can configure the VM. After the VM is configured, set the password for the `mib` user as the very last task.

Reply:
Then can I set a default password on the MiB VM, and ask the user to change it when they receive the vm?

Reply:
I tried it out with different password from the pve node sudo password but the script will fail with this error `fatal: [mib1]: FAILED! => {"msg": "Incorrect sudo password"}`

Reply:
I have to shift the `ansible_become_pass` variable for pve-node to `roles/create_vm/vars/main.yaml` file for the script to work. The variable is shifted there because for the creation of vm, need to become sudo user for the pve node.

Reply:
Like the DB credentials, a unique password should be set for each VM. The user should be advise to reset the password after receiving the appliance (add this to the documentation).

> I have to shift the ansible_become_pass variable for pve-node to roles/create_vm/vars/main.yaml file for the script to work. The variable is shifted there because for the creation of vm, need to become sudo user for the pve node.

Can you check if the VM's default behaviour permits passwordless sudo? I believe that's the case. If it's not, then what you could do is to first set the `mib` user password (via cloud-init) to `ansible_become_pass`. After the VM is fully configured, set the password to the deployment-specific one.

Reply:
> Like the DB credentials, a unique password should be set for each VM. The user should be advise to reset the password after receiving the appliance (add this to the documentation).

Ok, I will use ansible vault to set a common password for all MiB VM. 

> Can you check if the VM's default behaviour permits passwordless sudo? I believe that's the case. If it's not, then what you could do is to first set the mib user password (via cloud-init) to ansible_become_pass. After the VM is fully configured, set the password to the deployment-specific one.

Yup the VM, default behaviour is passwordless sudo. I did a test with the `ask-become-pass` flag and it is able to deploy successfully without any issue. I will remove the `ansible_become_pass` variable.

Reply:
1. I do have a question, if we are going to set the default user as "mib" and also the default password, should we put it in the `group_vars/all.yml` file instead? Since it will be the same for all the MiB VMs being created and set up(before delivery to the end user).

1. Since we will be setting a default password and advising the user to change the password upon receiving the MiB VM, do we have the use ansible vault to encrypt it? (Because those major appliances in the market, their default password can be found easily online)

Reply:
1. Ideally, the `mib` user in each VM should be configured with a unique password. This can be specified as a vaulted variable in the deployment's hostvar. There are 2 ways to handle deployments:
   1. Set a default password for all VMs. Make the client reset the password as part of the SOP. Add this to the documentation.
   1. Set a unique password for each deployment and leave it to the client to reset the password. I personally think this is safer but I leave it up to you.
1. Always put passwords, secrets and sensitive info in a vault. Always practice good security hygiene by default. "Major appliances" such as home routers are in a different category because they can be reset to a default password if the config is messed up.

Reply:
> 1. Ideally, the `mib` user in each VM should be configured with a unique password. This can be specified as a vaulted variable in the deployment's hostvar. There are 2 ways to handle deployments:
>    
>    1. Set a default password for all VMs. Make the client reset the password as part of the SOP. Add this to the documentation.
>    2. Set a unique password for each deployment and leave it to the client to reset the password. I personally think this is safer but I leave it up to you.

I will go with the first approach because the 2nd approach will be quite tedious during setup as we will need to constantly generate a new unique vault password. I will set the MiB VM password to expire using `passwd -e mib` so that it will force the user to key in a new password on the next login

> 2. Always put passwords, secrets and sensitive info in a vault. Always practice good security hygiene by default. "Major appliances" such as home routers are in a different category because they can be reset to a default password if the config is messed up.

Got it! Thanks

Reply:
Ok but the deployment SOP must include telling the user to reset the password. If not an attacker at some point can reset the password.

Reply:
I will just expire the password, so user definitely has to set a new password at first login.